src/Eloquent/Builder.php

```diff
@@ -202,56 +198,37 @@ public function raw($value = null)
         return $results;
     }
 
-    /**
-     * Attempt to create the record if it does not exist with the matching attributes.
-     * If the record exists, it will be returned.
-     *
-     * @param  array $attributes The attributes to check for duplicate records
-     * @param  array $values     The attributes to insert if no matching record is found
-     */
-    public function createOrFirst(array $attributes = [], array $values = []): Model
+    public function firstOrCreate(array $attributes = [], array $values = [])
```

Comment:
I'm not actually suggesting this, but it's amusing to me that this entire method body could be reduced to:

```php
return (clone $this)->where($attributes)->first() ??
    $this->getConnection()->getSession()?->isInTransaction()
        ? $this->create(array_merge($attributes, $values))
        : $this->createOrFirst($attributes, $values);
```

![](https://i.kym-cdn.com/photos/images/original/001/330/404/f38.png)